blackbox/docs/config/secure.rst

```diff
@@ -0,0 +1,73 @@
+.. highlight:: sh
+
+.. _conf-secure-settings:
+
+===============
+Secure Settings
+===============
+
+Some settings are sensitive as they might store passwords, access keys or 
+other similar information. This type of data needs more protection than just 
+filesystem permissions and even more, should not be available in plain text.
+
+For these instances, we provide a keystore and the ``crate-keystore`` tool to 
+manage the settings in the keystore.
+
+.. NOTE::
+
+    Changes to the keystore take affect only after restarting CrateDB
+
+.. NOTE::
+
+    The CrateDB keystore currently provides obfuscation, but in the future 
+    password protection will be added.
+
+.. NOTE::
+
+    Not all settings are read from the keystore. Consult the documentation for 
+    each setting to see if it is secure setting.
+
+
+Creating the keystore
+---------------------
+To create the crate.keystore, use the ``create`` command:
+
+.. code-block:: sh
+
+    sh$ ./bin/crate-keystore create
+
+The file ``crate.keystore`` will be created alongside ``crate.yml``.
+
+Listing settings in the keystore
+--------------------------------
+To list all the settings in the keystore, use the ``list`` command:
+
+.. code-block:: sh
+
+    bsh$ ./bin/crate-keystore list
+
+Adding string settings
+----------------------
+Sensitive string settings, like authentication credentials for cloud plugins, 
```

Comment:
add newline above this

Reply:
please explain what a "string setting" is (as opposed to "a setting")

what is a "cloud plugin"?

Reply:
dropped both (we don't talk about the cloud plugins anywhere else like ES does so no need to mention anything like that)